### PR TITLE
[pNOT MERGE] only for Test the Master Code

### PR DESCRIPTION
**Pull request by HassanDhariwal**

### CODE REVIEW

File: @student.save [hunk 1]
```diff
@@ -0,0 +1,26 @@
+
+[1mFrom:[0m /home/hassan/project/student_managment/app/controllers/students_controller.rb:22 StudentsController#create:
+
```
Comment (abdulrehman3222) [inline @student.save:3]: Remove this file in the next push

File: [:student][:father_name], [hunk 2]
```diff
@@ -0,0 +1,96 @@
+
+                   SSUUMMMMAARRYY OOFF LLEESSSS CCOOMMMMAANNDDSS
```
Comment (abdulrehman3222) [inline [:student][:father_name],:2]: Remove this file

File: app/controllers/students_controller.rb [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  end
+
+  def show
+    # binding.pry
+    @student = Student.find_by_id(params[:id])
+  end
+
+  def new
+    # binding.pry
+    @student = Student.new({:name => 'Enter your name', :father_name => 'Enter your Father Name ', age:"18"})
```
Comment (abdulrehman3222) [inline app/controllers/students_controller.rb:15]: @student = Student.new 

why you are passing these values of ENTER your Name

File: app/controllers/students_controller.rb [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+      )
+      redirect_to @student
+    else
+      flash[:error] = "Student not found."
+      render 'edit'
+    end
+  end
+
+  def destroy
+    @student = Student.find_by_id(params[:id])
```
Comment (abdulrehman3222) [inline app/controllers/students_controller.rb:63]: put it between line number 62 and 63
#TODO: Will set by Call back after complete the code

File: app/views/students/edit.html.erb [hunk 5]
```diff
@@ -0,0 +1,38 @@
+
+<div class="students Edit">
+<h2>Update Student</h2>
+
+
+<!-- <= form_with url: "/students/:id/edit", method: :post do |f| %>  -->
```
Comment (abdulrehman3222) [inline app/views/students/edit.html.erb:6]: commented code need to removed

File: app/views/students/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    padding-right: 40px; /* Adjust the desired amount of space */
+  }
+</style>
+
+<div class="students index">
+    <!-- <= link_to("Create a recode", '/students/new', :class => 'action edit' ) %> -->
+    <%= link_to("Create a recode", new_student_path, :class => 'action edit' ) %>
+<table class="table">
+	<thead>
+		<tr>
```
Comment (abdulrehman3222) [inline app/views/students/index.html.erb:15]: set the inteneration of this code

File: app/views/students/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    		<th>Blood_group</th>
+    		<th>Show</th>	
+   			<th>Edit</th>
+    		<th>Delete</th>
+  		</tr>
+  	</thead>
+  	<tbody>	
+  	<% @students.each do |student|%>
+  		<tr>
+  			<td><%= student.id %></td>
```
Comment (abdulrehman3222) [inline app/views/students/index.html.erb:31]: same comment as above

File: config/routes.rb [hunk 8]
```diff
@@ -1,5 +1,26 @@
...
+
+
+  # get 'student/new'
+  # get 'student/show's
+  # post 'student/create'
+
+
+
+# get '/hotels', to: 'hotels#index'
+# get '/hotels/:id', to: 'hotels#show'
```
Comment (abdulrehman3222) [inline config/routes.rb:18]: remove extra routes not needed any more

File: db/migrate/20230527101823_change_column_to_students.rb [hunk 9]
```diff
@@ -0,0 +1,5 @@
+class ChangeColumnToStudents < ActiveRecord::Migration[5.1]
+  def change
+    change_column("students", "age", :integer)
```
Comment (abdulrehman3222) [inline db/migrate/20230527101823_change_column_to_students.rb:3]: in the case of change use the up and down

File: db/seeds.rb [hunk 10]
```diff
@@ -5,3 +5,44 @@
...
+# 2nd recode
+# s_class = StudentClass.new(section_name: "A", teacher_name: "Ali", student_name: "abubakr" )
+# 3rd recode
+# s_class = StudentClass.create(section_name: "C", teacher_name: "ZAiN", student_name: "ABR" )
+# 4th recode
+# s_class = StudentClass.create(section_name: "A", teacher_name: "farooq", student_name: "ali" )
+# 5th recode
+# s_class = StudentClass.new(section_name: "A", teacher_name: "farooq_ali", student_name: "ali" )
+# 2.7.1 :043 > s_class.save
+# 5th recode not Write this
```
Comment (abdulrehman3222) [inline db/seeds.rb:35]: keep only commands here you copy past the whole console 
SHABASH

File: tudent = Student.new(  [hunk 11]
```diff
@@ -0,0 +1,52 @@
+
+[1mFrom:[0m /home/hassan/project/student_managment/app/controllers/students_controller.rb:22 StudentsController#create:
+
+    [1;34m21[0m: [32mdef[0m [1;34mcreate[0m
```
Comment (abdulrehman3222) [inline tudent = Student.new( :4]: Delete this file as well

**Review by abdulrehman3222 — CHANGES_REQUESTED** [review]
Need to change

File: @student.save (hunk 1 above)
Comment (HassanDhariwal) [reply @student.save:3]: done

File: [:student][:father_name], (hunk 2 above)
Comment (HassanDhariwal) [reply [:student][:father_name],:2]: done

File: app/controllers/students_controller.rb (hunk 4 above)
Comment (HassanDhariwal) [reply app/controllers/students_controller.rb:63]: done

File: app/views/students/edit.html.erb (hunk 5 above)
Comment (HassanDhariwal) [reply app/views/students/edit.html.erb:6]: done

File: config/routes.rb (hunk 8 above)
Comment (HassanDhariwal) [reply config/routes.rb:18]: done

File: app/views/students/index.html.erb (hunk 7 above)
Comment (HassanDhariwal) [reply app/views/students/index.html.erb:31]: done

File: tudent = Student.new(  (hunk 11 above)
Comment (HassanDhariwal) [reply tudent = Student.new( :4]: done

File: app/controllers/students_controller.rb (hunk 3 above)
Comment (HassanDhariwal) [reply app/controllers/students_controller.rb:15]: done

File: db/seeds.rb (hunk 10 above)
Comment (HassanDhariwal) [reply db/seeds.rb:35]: done

**Pull request closed without merging**